app/backends/workflow-orchestration.ts

```diff
@@ -80,27 +80,55 @@ async function _handleWorkItemResults(
   }
 }
 
+/**
+ * Creates a work item that uses all the output of the previous step.
+ * 
+ * @param tx - The database transaction
+ * @param currentWorkItem - The current work item
+ * @param nextStep - the next step in the workflow
+ * @param results - an array of paths to STAC catalogs
+ */
+async function createAggregatingWorkItem(
+  tx: Transaction, currentWorkItem: WorkItem, nextStep: WorkflowStep, results: string[],
+): Promise<void> {
+  // TODO use ALL outputs from the prior step here (HARMONY-961)
+  // just use the first result of the current output for now
+  const newWorkItem = new WorkItem({
+    jobID: currentWorkItem.jobID,
+    serviceID: nextStep.serviceID,
+    status: WorkItemStatus.READY,
+    stacCatalogLocation: results[0],
+    workflowStepIndex: nextStep.stepIndex,
+  });
+
+  await newWorkItem.save(tx);
+}
+
 /**
  * Creates the next work items for the workflow based on the results of the current step
  * @param tx - The database transaction
  * @param currentWorkItem - The current work item
  * @param nextStep - the next step in the workflow
  * @param results - an array of paths to STAC catalogs
  */
-async function _createNextWorkItems(
+async function createNextWorkItems(
   tx: Transaction, currentWorkItem: WorkItem, nextStep: WorkflowStep, results: string[],
 ): Promise<void> {
-  // Create a new work item for each result using the next step
-  for await (const result of results) {
-    const newWorkItem = new WorkItem({
-      jobID: currentWorkItem.jobID,
-      serviceID: nextStep.serviceID,
-      status: WorkItemStatus.READY,
-      stacCatalogLocation: result,
-      workflowStepIndex: nextStep.stepIndex,
-    });
+  if (nextStep.hasAggregatedOutput) {
+    return createAggregatingWorkItem(tx, currentWorkItem, nextStep, results);
+  } else {
+    // Create a new work item for each result using the next step
+    for await (const result of results) {
```

Comment:
Not for your PR to fix, but it strikes me that for no granule limit, we're probably going to want/need some sort of bulk insert (`INSERT INTO ... SELECT ...`)

Reply:
In general it'd be good to start making note of any time we're iterating over all results in prep for that.

Reply:
I added a FIXME comment with reference to NO GRANULE LIMIT ticket (HARMONY-276)